### PR TITLE
HDDS-4660. Merge OMTransactionInfo with SCMTransactionInfo

### PR DESCRIPTION
**Pull request by bshashikant**
## What changes were proposed in this pull request?
Refactoring change for OMTransactionInfo/SCMTransactionInfo

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-4660

## How was this patch tested?

Refactoring change, Existing unit tests should suffice.

### CODE REVIEW

File: hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/DBStoreHAManager.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import org.apache.hadoop.hdds.utils.db.Table;
+
+/**
+ * Interface defined for getting HA related specific info from DB for SCM and
+ * OM.
+ */
+public interface DBStoreHAManager {
+
+  default Table<String, TransactionInfo> getTransactionInfoTable() {
```
Comment (amaliujia) [inline hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/DBStoreHAManager.java:28]: +1 I was actually stuck on this one: I wasn't sure how to deal with this class in OM.

File: hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/TransactionInfo.java [hunk 2]
```diff
@@ -159,9 +186,8 @@ public Builder setTransactionIndex(long tIndex) {
       return this;
     }
 
-    public OMTransactionInfo build() {
-      return new OMTransactionInfo(currentTerm, transactionIndex);
+    public TransactionInfo build() {
+      return new TransactionInfo(currentTerm, transactionIndex);
     }
-
```
Comment (amaliujia) [inline hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/TransactionInfo.java:165]: nit: undo this line

**Review by amaliujia — APPROVED** [review]
LGTM 
This refactoring is straightforward.

**Comment by bshashikant** [comment]
Thanks @amaliujia for the review.

**Pull request merged**